lib/practice_exercises.rb

```diff
@@ -1,13 +1,47 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: 0(1)
 def remove_duplicates(list)
-  raise NotImplementedError, "Not implemented yet"
+  i = 0
+  j = 1
+  until list[j] == nil
+    if list[i] == list[j]
+      list.delete_at(list[j])
```

Comment:
`delete_at` removes an element by shifting all subsequent elements left by one index.  So it's an O(n) method.  Since it's nested inside a loop running n times, `remove_duplicates` is an O(n<sup>2</sup>) method.